packages/instantsearch.js/src/__tests__/common.test.tsx

```diff
@@ -25,11 +43,30 @@ import {
   hits,
   index,
   rangeInput,
+  hitsPerPage,
+  toggleRefinement,
 } from '../widgets';
 
 createHierarchicalMenuTests(({ instantSearchOptions, widgetParams }) => {
+  const hierarchicalMenuURL = connectHierarchicalMenu<{
+    container: HTMLElement;
+  }>((renderOptions) => {
+    renderOptions.widgetParams.container.innerHTML = `
+        <a
+          data-testid="HierarchicalMenu-link"
+          href="${renderOptions.createURL('value')}"
+        >
+          LINK
+        </a>
+      `;
+  });
```

Comment:
I'm still confused on why we need to create a dedicated component to test this. Can't this be tested directly by checking that the links generated by the built-in widgets work properly?

Reply:
most widgets don't have links, which is why for consistency all widgets are tested with a custom component.

Reply:
Then maybe this shouldn't go in the setup for widgets, and instead we need a separate common test suite for connectors/Hooks.

Reply:
Could be interesting, although now the routing tests also use the widgets that get mounted alongside it. How would you see it simplified?

Reply:
If widgets are creating links with `createURL` then I'd test the links generated by the widget (in the `routing` suite as you're doing now).

If they're not, but their connector expose `createURL`, then I'd add `tests/common/connectors` for connector tests with a `routing` test suite as the one for widgets.

Then in packages (e.g., `instantsearch.js`) I'd execute the `create*` functions twice, once with the setup for widgets, once with the setup for connectors.

Reply:
https://algolia.atlassian.net/browse/FX-2468